browser/brave_browser_process_impl.cc

```diff
@@ -262,9 +256,11 @@ BraveBrowserProcessImpl::debounce_component_installer() {
 
 brave_shields::HTTPSEverywhereService*
 BraveBrowserProcessImpl::https_everywhere_service() {
-  if (!https_everywhere_service_)
+  if (!created_https_everywhere_service_) {
```

Comment:
but why?

Reply:
this is pretty standard for lazy init and you can find many examples in browser_process_impl.cc. I just changed it to match upstream conventions

Reply:
I'm pretty sure those four upstream occurences of `bool` in `browser_process_impl.h` are leftovers from ancient times (mb `scoped_ptr` was not convertible to bool?) You can see that all new additions to browser_process_impl simply check `unique_ptr`s, no additional flags are really needed.

Reply:
I don't remember why I bothered doing it in the first place, maybe some test issue?

Reply:
I feel like there was some reason I did this, but it was so long ago I can't remember. If it doesn't cause any problems I'm fine with undoing it